tests/test.cpp

```diff
@@ -1,42 +1,167 @@
-#include "../Jsonparser.cpp"
+#include "Jsonparser.h"
+#include "Szorny.h"
+#include "Kalandor.h"
 #include "gtest/gtest.h"
 
 
 using namespace std;
 
-
-TEST(Jsontest, Fajlvaltozo) {
+TEST(Jsonparsertest, Fajlvaltozo) {
 	std::ifstream f("Hosarkany.json");
 	Jsonparser beolvasas(f);
 	EXPECT_EQ(beolvasas.getErtek("name"), "Hosarkany");
 	EXPECT_EQ(beolvasas.getErtek("hp"), "300");
 	EXPECT_EQ(beolvasas.getErtek("dmg"), "30");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.4");
 }
 
-TEST(Jsontest, Szoveg) {
+TEST(Jsonparsertest, Szoveg) {
 	std::string szoveg = "Sotetvarazslo.json";
 	Jsonparser beolvasas(szoveg);
 	EXPECT_EQ(beolvasas.getErtek("name"), "Sotetvarazslo");
 	EXPECT_EQ(beolvasas.getErtek("hp"), "250");
 	EXPECT_EQ(beolvasas.getErtek("dmg"), "40");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.0");
 }
```

Comment:
a Sotetvarazslo.json nem kéne hogy hibát dobjon abból kifolyólag, hogy a 'hp' kulcsnak hiányzik egy aposztrófja?

Reply:
igazából az első 3 teszt még az előző körből van, de lekezeli a "nem tökéletes" fájlokat is(tanár úr kérésére ki kellett próbálni).

Reply:
nem arra kell felkészíteni, hogy ne zavarja a több/kevesebb space és a kulcsok-érték párok sorrendje?

Reply:
Azt szeretném látni, hogy a program képes elbánni olyan JSON fájllal, ami pl.
"name" : "error,
vagy
"name : error",
vagy
"name : "error",
sort tartalmaz.
-Hatwagner Tanár Úr ezen kéréseinek eleget téve lett olyan json fájl,amilyen. Ezekre az esetekre kellett felkészíteni a kódot az előző körben.

Reply:
akkor ez is rendben, én tévedtem